src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final int MILLIS = 300;
+
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+        return CompletableFuture.supplyAsync(() -> getUserDetails(userId), executor);
+    }
+
+    private UserData getUserDetails(String userId) {
+        try {
+            Thread.sleep(MILLIS);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Failed to retrieve details for userId: " + userId, e);
+        }
+
+        UserData userData = new UserData(userId, "Details for " + userId);
+        cache.put(userId, userData);
+        return userData;
```

Comment:
The `getUserDetails` method uses `Thread.sleep` which is not recommended in asynchronous processing as it blocks the current thread. Instead, consider simulating a delay using a scheduled executor or other non-blocking mechanisms.

Reply:
I've added `Thread.sleep()` to simulate a working process.

* ` The processing can be a simulated task, like fetching user data from a database (mock this with a delay).`